plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java

```diff
@@ -422,7 +422,7 @@ public List<Long> getPastTableVersions(TrinoFileSystem fileSystem, String transa
                 }
             }
             catch (IOException e) {
-                if (isFileNotFoundException(e)) {
+                if (e instanceof FileNotFoundException) {
```

Comment:
Now that this is a separate exception, use a different catch clause.

Reply:
The `isFileNotFoundException` method was verifying if the exception thrown was of type `java.io.FileNotFoundException` or if the error message contained the string `The specified key does not exist`.

As we have mapped the error message to `java.io.FileNotFoundException`, it is now enough to check if the exception is an instance of `FileNotFoundException` using the `e instanceof FileNotFoundException syntax`.

Am I missing something here?

```java
    public static boolean isFileNotFoundException(IOException e)
    {
        if (e instanceof FileNotFoundException) {
            return true;
        }
        if (e.getMessage().contains("The specified key does not exist")) {
            return true;
        }
        return false;
    }
```

Reply:
I think that @electrum points out a handling similar to this one:

```
            catch (FileNotFoundException e) {
                // no longer exists, break iteration
                return null;
            }
            catch (IOException e) {
                throw new UncheckedIOException(e);
            }
```

Reply:
Thanks @findinpath | @electrum, I got the point :)